test flake: TestNamespaceScheduler/GVRs_are_removed,_and_then_quickly_re-added_to_a_new_workspace

```
 --- FAIL: TestNamespaceScheduler (0.63s)
    fixture.go:109: shared kcp server will target configuration "/home/runner/work/kcp/kcp/.kcp/admin.kubeconfig"
    fixture.go:240: Created organization workspace root:e2e-org-tcfqb
    --- FAIL: TestNamespaceScheduler/GVRs_are_removed,_and_then_quickly_re-added_to_a_new_workspace (30.71s)
        fixture.go:299: Created Universal workspace root:e2e-org-tcfqb:e2e-workspace-j726x
        controller_test.go:255: Starting namespace expecter
        controller_test.go:269: Set up clients for test after 484.320534ms
        controller_test.go:270: Starting test...
        controller_test.go:147: Create a ready WorkloadCluster, and keep it artificially ready
        controller_test.go:154: 
            	Error Trace:	controller_test.go:154
            	            				controller_test.go:272
            	Error:      	Condition never satisfied
            	Test:       	TestNamespaceScheduler/GVRs_are_removed,_and_then_quickly_re-added_to_a_new_workspace
FAIL
FAIL	github.com/kcp-dev/kcp/test/e2e/reconciler/namespace	45.680s
```

https://github.com/kcp-dev/kcp/runs/6497673065?check_suite_focus=true

--- a/test/e2e/reconciler/namespace/controller_test.go
+++ b/test/e2e/reconciler/namespace/controller_test.go
@@ -139,8 +139,10 @@ func TestNamespaceScheduler(t *testing.T) {
 			work: func(ctx context.Context, t *testing.T, server runningServer) {
 				crdClusterClient, err := apiextensionsclient.NewClusterForConfig(server.DefaultConfig(t))
 				require.NoError(t, err, "failed to construct apiextensions client for server")
+
 				dynamicClusterClient, err := dynamic.NewClusterForConfig(server.DefaultConfig(t))
 				require.NoError(t, err, "failed to construct dynamic client for server")
+
 				kubeClusterClient, err := kubernetes.NewClusterForConfig(server.DefaultConfig(t))
 				require.NoError(t, err, "failed to construct kubernetes client for server")
 
@@ -151,12 +153,7 @@ func TestNamespaceScheduler(t *testing.T) {
 				}
 				cluster, err = server.kcpClient.WorkloadV1alpha1().WorkloadClusters().Create(ctx, cluster, metav1.CreateOptions{})
 				require.NoError(t, err, "failed to create cluster")
-				require.Eventually(t, func() bool {
-					cluster, err = server.kcpClient.WorkloadV1alpha1().WorkloadClusters().Get(ctx, cluster.Name, metav1.GetOptions{})
-					conditions.MarkTrue(cluster, "Ready")
-					cluster, err = server.kcpClient.WorkloadV1alpha1().WorkloadClusters().UpdateStatus(ctx, cluster, metav1.UpdateOptions{})
-					return err == nil
-				}, wait.ForeverTestTimeout, time.Millisecond*100)
+
 				go wait.UntilWithContext(ctx, func(ctx context.Context) {
 					patchBytes := []byte(fmt.Sprintf(`[{"op":"replace","path":"/status/lastSyncerHeartbeatTime","value":%q}]`, time.Now().Format(time.RFC3339)))
 					_, err := server.kcpClient.WorkloadV1alpha1().WorkloadClusters().Patch(ctx, cluster.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{}, "status")

